src/deploy/hasher_segments.js

```diff
@@ -28,13 +28,14 @@ const fileNormalizerCtor = ({ assetType = 'file' }) =>
 // A writable stream segment ctor that normalizes file paths, and writes shaMap's
 const manifestCollectorCtor = (filesObj, shaMap, { statusCb, assetType }) => {
   if (!statusCb || !assetType) throw new Error('Missing required options')
-  return objWriter((fileObj, _, cb) => {
+  return flushWriteStream.obj((fileObj, _, cb) => {
     filesObj[fileObj.normalizedPath] = fileObj.hash
 
     // We map a hash to multiple fileObj's because the same file
     // might live in two different locations
 
     if (Array.isArray(shaMap[fileObj.hash])) {
+      // eslint-disable-next-line fp/no-mutating-methods
```

Comment:
This code relies on streaming for performance reasons (hashing many files). Streaming relies on event handlers, which are hard to use in a purely functional way, which is why we need some mutation here.

There is a probably a way to do this in an incremental/performant way without mutation, but this could potentially break the logic.